src/logging.c

```diff
@@ -104,47 +173,149 @@ void rcutils_logging_console_output_handler(
-  char * message_buffer = buffer;
+  // Declare variables that will be needed for cleanup ahead of time.
+  char static_output_buffer[1024];
+  char * output_buffer = NULL;
+
+  // Start with a fixed size message buffer and if during message formatting we need longer, we'll
+  // dynamically allocate space.
+  char static_message_buffer[1024];
+  char * message_buffer = static_message_buffer;
+
   int written;
   {
     // use copy of args to keep args for potential later user
     va_list args_clone;
     va_copy(args_clone, *args);
-    written = vsnprintf(buffer, sizeof(buffer), format, args_clone);
+    written = vsnprintf(static_message_buffer, sizeof(static_message_buffer), format, args_clone);
     va_end(args_clone);
   }
   if (written < 0) {
     fprintf(stderr, "failed to format message: '%s'\n", format);
     return;
   }
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
-  if ((size_t)written >= sizeof(buffer)) {
+  if ((size_t)written >= sizeof(static_message_buffer)) {
     // write was incomplete, allocate necessary memory dynamically
-    size_t buffer_size = written + 1;
-    void * dynamic_buffer = allocator.allocate(buffer_size, allocator.state);
-    written = vsnprintf(dynamic_buffer, buffer_size, format, *args);
-    if (written < 0 || (size_t)written >= buffer_size) {
+    size_t message_buffer_size = written + 1;
+    void * dynamic_message_buffer = allocator.allocate(message_buffer_size, allocator.state);
+    if (!dynamic_message_buffer) {
+      fprintf(stderr, "failed to allocate buffer for message\n");
+      return;
+    }
+    written = vsnprintf(dynamic_message_buffer, message_buffer_size, format, *args);
+    message_buffer = (char *)dynamic_message_buffer;
+    if (written < 0 || (size_t)written >= message_buffer_size) {
       fprintf(
         stderr,
         "failed to format message (using dynamically allocated memory): '%s'\n",
         format);
-      return;
+      goto cleanup;
     }
-    message_buffer = (char *)dynamic_buffer;
   }
 
-  if (!location) {
-    fprintf(stream, "[%s] [%s]: %s\n", severity_string, name, message_buffer);
-  } else {
-    fprintf(
-      stream, "[%s] [%s]: %s (%s() at %s:%zu)\n", severity_string, name, message_buffer,
-      location->function_name, location->file_name, location->line_number);
+  // Process the format string looking for known tokens.
+  const char token_start_delimiter = '{';
+  const char token_end_delimiter = '}';
+  // Start with a fixed size output buffer and if during token expansion we need longer, we'll
+  // dynamically allocate space.
+  output_buffer = static_output_buffer;
+  memset(static_output_buffer, '\0', sizeof(static_output_buffer));
+  size_t output_buffer_size = sizeof(static_output_buffer);
+  const char * str = g_rcutils_logging_output_format_string;
+  size_t size = strlen(g_rcutils_logging_output_format_string);
+
+  // Walk through the format string and expand tokens when they're encountered.
+  size_t n = 0;
```

Comment:
`n` seems to be only used once within the loop. Can it be declared there instead?

Reply:
Yep, I forgot that we prefer that f07539581e94dfb32c02c6e849fc44c30e7e3ca8